File: src/main/java/org/dedda/bratwurst/lang/Pop.java
package org.dedda.bratwurst.lang;

import org.dedda.bratwurst.lang.scope.Scope;

/**
 * Created by dedda on 12/8/15.
 *
 * @author dedda
 */
public class Pop extends BWInstruction {

    private String variableName;

    public Pop(int lineNumber, String variableName) {
        super(lineNumber);
        this.variableName = variableName;
    }

    @Override
    public void run(Scope scope) {
        scope.setVariable(new BWVariable(variableName, scope.pop()));
    }

    public String getVariableName() {
        return variableName;
    }
}

File: src/main/java/org/dedda/bratwurst/lang/Pop.kt
package org.dedda.bratwurst.lang

import org.dedda.bratwurst.lang.scope.Scope

/**
 * Created by dedda on 12/8/15.
 *
 * @author dedda
 */
class Pop(lineNumber: Int, val variableName: String) : BWInstruction(lineNumber) {
    override fun run(scope: Scope) {
        val value: BWExpression = scope.pop()
        val variable = BWVariable(variableName, value)
        variable.run(scope)
        scope.setVariable(variable)
    }

}